app/chat/page.tsx

```diff
@@ -10,7 +10,9 @@ interface ChatPageProps {
 
 export default async function ChatPage({ searchParams }: ChatPageProps) {
   const id = generateUUID();
-  const initialMessage = (await searchParams)?.q as string;
+  const params = await searchParams;
+  // Use only 'q' for the initial message, regardless of source (agent or user).
+  const initialMessage = params?.q as string;
```

Comment:
are these changes necessary?
- what `problem` are they solving or `value` is being added with `extra lines of code`?

I suggest `git checkout origin/main app/chat/page.tsx`